### PR TITLE
LPS-113561 Avoid use of Liferay.provide in portal-settings-authentication-opensso-web

### PR DESCRIPTION
**Pull request by julien**
As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561) we eventually want to deprecate the `Liferay.provide` function.

To test this change:

- From the "Global Menu" click on the "Control Panel" tab
- Click on "Configuration > Instance Settings"
- Scroll down to the "Security" section and click on "SSO"
- Click on the "OpenSSO" tab
- Click on On the top right kebabmenu next to "OpenSSO"
- Click on "Test OpenSSO configuration"

As a result, a modal window should be opened and depending of your settings it should either tell you that the connection to the OpenSSO server was successful or that it failed, and there should be no JS errors in the browser's console

### CODE REVIEW

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
@wincent in this case I was wonderning if we couldn't do something like this:

```js
<aui:script>
window['<portlet:namespace />testOpenSSOSettings'] = function () {
  var A = AUI();
  A.use('aui-io-plugin-deprecated','liferay-util-window', function() {
    var data = {};
     ...
  });
};
</aui:script>
```

**Comment by wincent** [comment]
As per Slack, we're going to explore getting rid of the `liferay-util-window` (`getWindow`) dependency. Disclaimer: I have no idea what `A.Plugin.IO` does and how easily it will be to replace it.

**Comment by julien** [comment]
Here's a screenshot of how it looked before:

before

And how it looks after:

after

File: modules/apps/portal-settings/portal-settings-authentication-opensso-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/opensso.jsp [hunk 1]
```diff
@@ -57,58 +57,66 @@ String version = openSSOConfiguration.version();
...
-				uri: url,
+		})
+		.catch(function (error) {
+			Liferay.Util.openToast({
+				message: Liferay.Language.get(
+					'an-unexpected-system-error-occurred'
+				),
+				title: Liferay.Language.get('error'),
+				type: 'danger',
 			});
```
Comment (wincent) [inline modules/apps/portal-settings/portal-settings-authentication-opensso-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/opensso.jsp:119]: Somebody buy this man a cigar!

Comment (wincent) [reply modules/apps/portal-settings/portal-settings-authentication-opensso-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/opensso.jsp:119]: cigar

**Comment by wincent** [comment]
Between your manual tests, and our CI, what could possibly go wrong?

nothing

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**